File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def find_divisor(number):
    
    count = 0
    
    for idx in range(1, int(number**(1/2)) + 1):
        if number % idx == 0:
            count += 1
            if ( (idx**2) != number) : 
                count += 1
                
            
    return count 

def solution(number, limit, power):
    answer = 0
    divisors = []
     
    for idx in range(1, number+1):
        divisors.append(find_divisor(idx))
        
    for i in range(len(divisors)):
        if divisors[i] > limit:
            divisors[i] = power
        answer += divisors[i]    
        
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 1;
        int[] weight = new int[number+1];
        
        for(int i = 2; i <= number; i++){
            int count = countDivNum(i);
            
            if(count > limit){
                weight[i-1] = power;
            } else{
                weight[i-1] = count;
            }
        }
        
        for(int i = 0; i < weight.length; i++){
            System.out.println(weight[i]);
            answer += weight[i];
        }
        
        return answer;
    }
    
    public int countDivNum(int number){
        
        int count = 2;
        
        for(int i = 2; i <= number / 2; i++){
            if(number % i == 0){
                count++;
            }
        }
        
        return count;
    }
}